.github/workflows/cli-tests.yaml

```diff
@@ -95,8 +100,14 @@ jobs:
           echo "NIX_PROFILES=$NIX_PROFILES" >> $GITHUB_ENV
           echo "NIX_SSL_CERT_FILE=$NIX_SSL_CERT_FILE" >> $GITHUB_ENV
           echo "PATH=$PATH" >> $GITHUB_ENV
-      - name: Run tests
+      - name: Run go tests
         run: go test -race -cover ./...
+      - name: Run shellspec tests (bash)
+        run: shellspec testscripts/shellspec/ -s /bin/bash -f document --no-warning-as-failure
+      - name: Run shellspec tests (zsh)
+        run: shellspec testscripts/shellspec/ -s /bin/zsh -f document --no-warning-as-failure
+      - name: Run shellspec tests (dash)
+        run: shellspec testscripts/shellspec/ -s /bin/dash -f document --no-warning-as-failure
```

Comment:
I'm not sure using `--no-warning-as-failure` is good (unless it's impossible to get around it). Right now some of the tests are failing because the output is to stderr is not checked. I think it's better to check it (ideally in a flexible way were it wont break if we change it). That way we can remove this flag and if stderr actually shows up we can catch it.